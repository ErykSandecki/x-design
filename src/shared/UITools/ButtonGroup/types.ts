// types
import { TTooltipProps } from '../../UI/Tooltip/Tooltip';
import { TIconProps } from '../../UI/Icon/Icon';

export type TButtonGroup = {
  disabled?: boolean;
  name: TIconProps['name'];
  onClick: TFunc;
  tooltip?: Omit<TTooltipProps, 'children'>;
};
