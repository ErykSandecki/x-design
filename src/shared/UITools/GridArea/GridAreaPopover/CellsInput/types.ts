export type TActiveCell = {
  columns: number;
  rows: number;
};
