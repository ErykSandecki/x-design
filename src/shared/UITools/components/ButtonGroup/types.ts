// types
import { TTooltipProps } from '../../../UI/components/Tooltip/Tooltip';
import { TIconProps } from '../../../UI/components/Icon/Icon';

export type TButtonGroup = {
  disabled?: boolean;
  name: TIconProps['name'];
  onClick: TFunc;
  tooltip?: Omit<TTooltipProps, 'children'>;
};
