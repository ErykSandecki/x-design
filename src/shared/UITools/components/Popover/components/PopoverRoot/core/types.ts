export type TContext = {
  setSelected: TFunc<[boolean]>;
};
