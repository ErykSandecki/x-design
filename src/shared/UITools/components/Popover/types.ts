// types
import { TIconProps } from 'shared/UI/components/Icon/Icon';

export type TPopoverData = {
  icon?: TIconProps['name'];
  onClick?: () => void;
  selected?: boolean;
  text: string;
};

export type TPopoverSeparator = { separator: boolean };

export type TPopoverItem = TPopoverData | TPopoverSeparator;

export type TPopover = {
  data: Array<TPopoverData | TPopoverSeparator>;
};
