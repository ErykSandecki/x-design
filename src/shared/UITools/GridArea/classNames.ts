// utils
import { composeClassNames } from 'utils';

export const className = 'GridArea';

export const classNames = composeClassNames(className, [className, 'fullWidth'] as const);
