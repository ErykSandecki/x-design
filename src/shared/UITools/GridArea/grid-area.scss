.GridArea {
  @include theme('neutral-3', background-color);
  @include theme('neutral-3', border, '1px solid');

  border-radius: 5px;
  box-sizing: border-box;
  cursor: pointer;
  height: 56px;

  &--full-width {
    width: 100%;
  }
}
