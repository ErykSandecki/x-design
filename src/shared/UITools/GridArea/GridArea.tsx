import { FC, useRef } from 'react';
import { noop } from 'lodash';

// components
import Box from '../../UI/Box/Box';
import Cells, { TCellsProps } from './Cells/Cells';
import GridAreaPopover, { TGridAreaPopoverProps } from './GridAreaPopover/GridAreaPopover';

// hooks
import { useOutsideClick, useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';

// styles
import styles from './grid-area.scss';

// types
import { E2EAttribute } from 'types';
import { TE2EDataAttributeProps } from '../../E2EDataAttributes/E2EDataAttribute';

export type TGridAreaProps = TCellsProps &
  Pick<TGridAreaPopoverProps, 'onBlurColumns' | 'onBlurRows' | 'onChangeColumns' | 'onChangeRows'> & {
    e2eValue?: TE2EDataAttributeProps['value'];
    fullWidth?: boolean;
    idContainer?: string;
  };

export const GridArea: FC<TGridAreaProps> = ({
  columns,
  e2eValue = '',
  fullWidth = false,
  idContainer,
  rows,
  ...restProps
}) => {
  const ref = useRef(null);
  const { selected, setSelected } = useOutsideClick([], ref, noop, idContainer);
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);

  return (
    <Box
      classes={{
        className: cx(className, classNamesWithTheme[className].name, [
          classNamesWithTheme[className].modificators.fullWidth,
          fullWidth,
        ]),
      }}
      e2eAttribute={E2EAttribute.gridArea}
      e2eValue={e2eValue}
      onClick={() => setSelected(true)}
      ref={ref}
    >
      <Cells columns={columns} rows={rows} />
      <GridAreaPopover
        columns={columns}
        ref={ref}
        rows={rows}
        selected={selected}
        setSelected={setSelected}
        {...restProps}
      />
    </Box>
  );
};

export default GridArea;
