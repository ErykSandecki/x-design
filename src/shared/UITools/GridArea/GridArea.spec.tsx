import { fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';

// components
import GridArea from './GridArea';

// store
import { configureStore } from 'store/store';

// types
import { E2EAttribute } from 'types';

// utils
import { customRender, getByE2EAttribute } from 'test';

describe('GridArea snapshots', () => {
  it('should render GridArea', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment } = customRender(
      <Provider store={store}>
        <GridArea columns={1} rows={1} />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render when exceed 10x10', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment } = customRender(
      <Provider store={store}>
        <GridArea columns={100} rows={100} />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render wner popover selected', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment, container } = customRender(
      <Provider store={store}>
        <GridArea columns={1} rows={1} />
      </Provider>,
    );

    // action
    fireEvent.click(getByE2EAttribute(container, E2EAttribute.gridArea));

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
