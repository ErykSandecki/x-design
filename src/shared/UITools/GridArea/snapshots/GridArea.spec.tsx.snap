// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridArea snapshots should render GridArea 1`] = `
<DocumentFragment>
  <div
    class="css-1jnxvbp GridArea GridArea GridArea--light"
    data-test-grid-area=""
  >
    <div
      class="css-1jnxvbp Cells Cells Cells--light"
      data-test-grid-area=""
      style="grid-template-columns: repeat(1, 1fr); grid-template-rows: repeat(1, 1fr);"
    >
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: 5px; border-bottom-right-radius: 5px; border-top-left-radius: 5px; border-top-right-radius: 5px;"
      />
      <small
        class="css-1jnxvbp Cells__sizes Cells__sizes--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        1 
        <span>
          x
        </span>
         1
      </small>
    </div>
    <div
      class="css-6wln18 Popover Popover--light"
      data-test-box=""
      data-test-popover="popover"
      style="height: 180px; width: 210px; left: -12.5px; top: -48.5px;"
    >
      children
    </div>
  </div>
</DocumentFragment>
`;

exports[`GridArea snapshots should render when exceed 10x10 1`] = `
<DocumentFragment>
  <div
    class="css-1jnxvbp GridArea GridArea GridArea--light"
    data-test-grid-area=""
  >
    <div
      class="css-1jnxvbp Cells Cells Cells--light"
      data-test-grid-area=""
      style="grid-template-columns: repeat(10, 1fr); grid-template-rows: repeat(10, 1fr);"
    >
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: 5px; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: 5px;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: 5px; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: none; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: none; border-bottom-right-radius: 5px; border-top-left-radius: none; border-top-right-radius: none;"
      />
      <small
        class="css-1jnxvbp Cells__sizes Cells__sizes--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        100 
        <span>
          x
        </span>
         100
      </small>
    </div>
    <div
      class="css-6wln18 Popover Popover--light"
      data-test-box=""
      data-test-popover="popover"
      style="height: 180px; width: 210px; left: -12.5px; top: -48.5px;"
    >
      children
    </div>
  </div>
</DocumentFragment>
`;

exports[`GridArea snapshots should render wner popover selected 1`] = `
<DocumentFragment>
  <div
    class="css-1jnxvbp GridArea GridArea GridArea--light"
    data-test-grid-area=""
  >
    <div
      class="css-1jnxvbp Cells Cells Cells--light"
      data-test-grid-area=""
      style="grid-template-columns: repeat(1, 1fr); grid-template-rows: repeat(1, 1fr);"
    >
      <div
        class="css-1jnxvbp Cells__cell Cells__cell--light"
        data-test-box=""
        style="border-bottom-left-radius: 5px; border-bottom-right-radius: 5px; border-top-left-radius: 5px; border-top-right-radius: 5px;"
      />
      <small
        class="css-1jnxvbp Cells__sizes Cells__sizes--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        1 
        <span>
          x
        </span>
         1
      </small>
    </div>
    <div
      class="css-6wln18 Popover Popover--light Popover--selected Popover--selected--light"
      data-test-box=""
      data-test-popover="popover"
      style="height: 180px; width: 210px; left: -12.5px; top: -48.5px;"
    >
      children
    </div>
  </div>
</DocumentFragment>
`;
